### PR TITLE
Call header_handler even if server returned 4xx response

### PR DESCRIPTION
**Pull request by ry023**
It seems to ignore `header_handler` when `CURLOPT_FAILONERROR` is enabled, so `highest(or lowest)_user_id` is't going to be stored if the UID not found.

It cause trouble such as when execute `useradd -r systemuser`.
In this case, `useradd` tries to resolve 1 to 1000 uid. and all of them should not be found. As a result, 1000 http requests are made.
In my environment, `useradd -r` takes 4,5 minutes.

I changed `inner_http_request` func:
- disable `CURLOPT_FAILONERROR`
- return `CURLE_HTTP_RETURNED_ERROR` manually when 4xx response returned by server.

### CODE REVIEW

File: stns.c [hunk 1]
```diff
@@ -289,13 +288,13 @@ static CURLcode inner_http_request(stns_conf_t *c, char *path, stns_response_t *
...
 
-  if (result != CURLE_OK) {
+  long code;
+  curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &code);
+  if (code >= 400) {
     syslog(LOG_ERR, "%s(stns)[L%d] http request failed: %s", __func__, __LINE__, curl_easy_strerror(result));
-    if (result == CURLE_HTTP_RETURNED_ERROR) {
-      long code;
-      curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &code);
-      res->status_code = code;
```
Comment (pyama86) [inline stns.c]: why don't you set status code when return code over 4xx?

Comment (ry023) [reply stns.c]: sorry, simply i forgot to set :bow:

**Comment by pyama86** [comment]
thanks!

**Comment by pyama86** [comment]
released
https://github.com/STNS/libnss/releases/tag/v2.2.3

**Pull request merged**